File: src/Labb1_task1/ReverseText.c
#include <stdio.h>
#include <stdlib.h>
void recursiveReverse()
{
    char ch;
    if((ch = getchar()) != '\n')
    {
        recursiveReverse();
    }
    putchar(ch);
}

void iterativeReverse()
{
    char* str = (char*)malloc(50*sizeof(char));
    int len = 0;
    char ch = 0;
    while(ch != '\n')
    {
        ch = getchar();
        *str = ch;
        str++;
        len++;
    }
    while(len >= 0)
    {
        putchar(*str);
        str--;
        len--;
    }
    printf("\n");
}

int main(void)
{
    printf("Enter your text to be reversed iteratively \n");
    iterativeReverse();
    printf("Enter your text to be reversed recursively \n");
    recursiveReverse();
    return 0;
}

File: src/Labb1_task2/ReverseText.java
package Labb1_task2;

import java.util.Scanner;
import java.util.Stack;

public class ReverseText
{
    // takes input from stdin and reverses it either iteratively with a stack or recursively through array logic
    public static void main(String[] args)
    {
        Scanner sys = new Scanner(System.in);

        System.out.println("Enter your text ");
        Stack stack = new Stack();
        String text = sys.nextLine();
        char[] arr = new char[text.length()];
        for(int i = 0; i < text.length(); i++)
            stack.push(text.charAt(i));

        for(int i = 0; i < text.length(); i++)
            arr[i] = text.charAt(i);

        System.out.println("Type 1 for iterative, else recursive");
        int i = sys.nextInt();
        if(i == 1)
            iterativeReverse(stack);
        else
            recursiveReverse(arr, arr.length - 1);

    }

    public static void iterativeReverse(Stack stack)
    {
        StringBuilder sb = new StringBuilder("");
        while(!stack.isEmpty())
            sb.append(stack.pop());

        System.out.println(sb.toString());
    }

    public static void recursiveReverse(char[] arr, int size)
    {
        if(size >= 0)
        {
            System.out.print(arr[size]);
            recursiveReverse(arr, --size);
        }
    }

}


